### DISCUSSION TITLE
Last 24h period

### DISCUSSION THREAD

**Discussion by hynek** [plausible/analytics#528] [Anything Else]
Hi,

I'm currently evaluating Plausible and I find the Today view not very useful, depending on the hour of the day. "Last 24h" would be much more useful.

Is there any chance you could add that? Or add hours to Custom range? Because as far as I can tell, I can only use whole days there.

**Comment by metmarkosaric** [comment]
thanks for the feedback @hynek! we do have "last 24h" stat [on the landing page](https://plausible.io/sites) of your Plausible account as it makes it easy to get a quick overview of all your sites and compare them. 

but for individual sites, today / yesterday is the expected behavior. it's pretty much industry standard in web analytics to look at today rather than last 24 hours. 

we plan to improve "today" to only show hours up to now and for comparison of today to be with the same amount of time rather than full day yesterday but there's no plan to remove it or replace it with 24 hours. 

we can look at if there's a way to get in it the custom range somehow if the request proves to be popular. thanks again!

**Reply by tejpochiraju** [reply]
@metmarkosaric - could you add "Yesterday" to the dropdown in this case? Below Today or Realtime would be ideal. 

I don't find Today super useful as most of customers/potentials are in the USA and we are in India (~12h ahead). As a result, I usually review stats the next morning. And I keep reaching for the custom range. This shortcut would help a good bit :-). 

![image](https://user-images.githubusercontent.com/9004891/108792686-dc3e7100-75a7-11eb-9a32-36fafa595e50.png)

**Reply by Vigasaurus** [reply]
You can select Today and then use the arrow keys, or the two arrows on the left of the drop-down to navigate adjacent periods. No need to use a custom range for something that close.

**Reply by tejpochiraju** [reply]
> You can select Today and then use the arrow keys, or the two arrows on the left of the drop-down to navigate adjacent periods. No need to use a custom range for something that close.

Wow, somehow did not notice the arrows. Thanks!

**Reply by simonw** [reply]
I came here to ask for a "Yesterday" filter too - I hadn't noticed those arrows either.

**Comment by chiragnd** [comment]
I searched this for a similar reason. I understand as @metmarkosaric mentioned, but it’s “slightly” confusing because the click out tells me a 24h number, but I cannot actually see _that_ number once I click; I’m left with either 12am to now, or selecting yesterday & today. So I think it would be great if I can drill down that last 24h number, or see a today number in the main screen that I jump into. 

But it feels ever so slightly unfulfilling (?) to start with a number and then have to look at completely different timescale once I click on it. Of course not critical - just a nice to have :)

**Comment by TkDodo** [comment]
I would like to pick this question up again, because it also confused me a bit: When you visit the dashboard in the morning, you are greeted with a bunch of red arrows for `unique visitors` and `total pageviews`, because obviously, they are smaller compared to "yesterday". But the current day only has 8 hours (at 8 in the morning), while yesterday had a full 24 hours, so that's rather not a fair comparison.

I understand that the `today` view is important to have, but I would think that an _additional_ `last 24 hours` view with comparison against the previous 24 hours would also be valuable. netlify does this in their analytics for example:

![Screenshot 2021-08-30 at 14 44 13](https://user-images.githubusercontent.com/1021430/131340906-ee651065-02f9-4379-8cca-07dda4266eff.png)

**Reply by metmarkosaric** [reply]
thanks @TkDodo! ideally, we will first fix the today comparison as it's broken in the current state (especially early in the day :) ). there's this thread about it https://github.com/plausible/analytics/discussions/344#discussioncomment-249990

**Reply by TkDodo** [reply]
oh nice, so even on the today view, the comparison would be for the last 24 hours? That would indeed get rid of the need for an additional 24h view I believe, but will be interesting to visualize ux wise :) Especially if the graph only shows today until right now, I think a hint that the comparison compares the last 24h would be good

**Reply by metmarkosaric** [reply]
the idea is to compare today to the exact same amount of time yesterday to make it a fair and accurate comparison. we haven't really got much further into how it will look like / work like

**Reply by TkDodo** [reply]
oh right, that also works. so at 8 in the morning, I would get a comparison from 00:00-08:00 yesterday ?

**Reply by metmarkosaric** [reply]
exactly!

**Reply by TkDodo** [reply]
sounds good, thanks for explaining 👍

**Reply by metmarkosaric** [reply]
you're welcome!

**Reply by TkDodo** [reply]
> the idea is to compare today to the exact same amount of time yesterday to make it a fair and accurate comparison.

one follow-up question to this please: Would it make sense to offer some sort of comparison with the same weekday from last week as opposed to just comparing it with yesterday always? For example, I am usually seeing waaay less traffic on my blog on weekends. So Saturday always has a big minus when compared with Friday, and Monday usually sees +100% on everything. Comparing Sunday to last week's Sunday would make more sense in these situations.

I understand that one design goal of plausible is to be simple to use compared with other analytics tools, so I'm totally fine if you feel that more and more filtering options would complicate things :)

**Reply by metmarkosaric** [reply]
the comparison will be a different feature. there is already some work in progress on the comparison so it will perhaps go live even before the fix of the last 24 hour period. you can follow the progress here https://github.com/plausible/analytics/discussions/532

**Comment by rosano** [comment]
Is there a way to have the last 24h via API? It seems to me you would need to make two calls for the last two days, filter hours for the 24h period, then aggregate. Is that correct?

**Reply by ukutaht** [reply]
The API has the same time periods as the dashboard. 'Last 24 hours' can be added to the API quite easily, though.

**Reply by rosano** [reply]
That would be amazing, even if undocumented. I'm hosted on plausible.io so I would just have to change the period string on my end.

**Comment by simonfrey** [comment]
Would put in UX here as argument as well. I see a number on the overview page that I want to further explore, but when clicking on the site, I have no option to see and dig into the same number.
If you want to stay with the "Today" setup, please at least also add a "Yesterday" shortcut

**Reply by metmarkosaric** [reply]
Thanks for the feedback @simonfrey! 

There are couple of easy ways to get to "yesterday":

- When you're viewing the stats for today, click on the left arrow button next to the date picker or press the left arrow key on your keyboard. Either one of these would take you to yesterday. 
- You can also hit the "E" key on your keyboard from any view. That should take you directly to yesterday as well. See all of our other keyboard shortcuts: https://plausible.io/docs/keyboard-shortcuts

**Comment by chiragnd** [comment]
Ah! Didn't know these existed, that's very helpful. Agree with @simonfrey
as well, since the number is on the dashboard,

Also, when I hit 'E', I'm taken to 28 Nov 1999 🤷‍♂️ All other shortcuts
seem to be working okay though.

-Cd

**Reply by metmarkosaric** [reply]
thanks for reporting @chiragnd! i've submitted an issue here: https://github.com/plausible/analytics/issues/1994

**Reply by metmarkosaric** [reply]
hi @chiragnd! this issue with the shortcut for yesterday has now been fixed

**Comment by HEAVYPOLY** [comment]
+ 1 for last 24 hours, last 30 days etc.  
Even better would be comparisons using last time period vs previous time period like how apple analytics does:
<img width="1136" alt="image" src="https://github.com/plausible/analytics/assets/24432677/4598645b-6e86-4203-bdb6-613b5f228569">

As it is now we only get a usable comparison at the 24 hour mark of each day:
<img width="1086" alt="image" src="https://github.com/plausible/analytics/assets/24432677/a0e0771b-fac6-430d-9c3a-83fd9bc18b83">
